ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -157,27 +150,72 @@ export const RuntimePanel = withCurrentWorkspace()(({workspace}) => {
                     closeOnClick
                     content={
                       <React.Fragment>
-                        <MenuItem style={styles.presetMenuItem}>General purpose analysis</MenuItem>
-                        <MenuItem style={styles.presetMenuItem}>Genomics analysis</MenuItem>
+                        {
+                          fp.filter(['displayName', 'General Analysis'], runtimePresets)
+                            // regular .map used here bc fp map won't provide index as an iteratee argument
+                            .map((preset, i) => {
+                              return <MenuItem
+                                  style={styles.presetMenuItem}
+                                  key={i}
+                                  onClick={() => {
+                                    // renaming to avoid shadowing
+                                    const presetDiskSize =
+                                        (preset.runtimeTemplate.gceConfig && preset.runtimeTemplate.gceConfig.bootDiskSize)
+                                        || (preset.runtimeTemplate.dataprocConfig && preset.runtimeTemplate.dataprocConfig.masterDiskSize);
+                                    const presetMachineName =
+                                        (preset.runtimeTemplate.gceConfig && preset.runtimeTemplate.gceConfig.machineType)
+                                        || (preset.runtimeTemplate.dataprocConfig && preset.runtimeTemplate.dataprocConfig.masterMachineType);
```

Comment:
Although unlikely to happen, it's possible that this can pull the presetDiskSize from gceConfig and presetMachineName from the dataprocConfig or vice versa.

Reply:
oh, that's true, I can make this not happen, but it'll take some `let`s to do it in a reasonable way